aot-build-time/src/main/java/com/dylibso/chicory/experimental/build/time/aot/Generator.java

```diff
@@ -101,14 +115,40 @@ public void generateMetaWasm() throws IOException {
                 wasmBytes,
                 section -> {
                     if (section.sectionId() == SectionId.CODE) {
+
+                        var source = ByteBuffer.wrap(((RawSection) section).contents());
+
                         var out = new ByteArrayOutputStream();
+                        var importFuncs = module.importSection().importCount();
                         int count = module.codeSection().functionBodyCount();
                         writeVarUInt32(out, count);
+                        readVarUInt32(source);
                         for (int i = 0; i < count; i++) {
-                            writeVarUInt32(out, 3); // function size in bytes
-                            writeVarUInt32(out, 0); // locals count
-                            out.write(OpCode.UNREACHABLE.opcode());
-                            out.write(OpCode.END.opcode());
+                            var funcId = importFuncs + i;
+                            if (this.interpretedFunctions.contains(funcId)) {
+
+                                // Copy over the original function body
+                                var bodySize = (int) readVarUInt32(source);
+                                writeVarUInt32(out, bodySize);
+                                var bodyBytes = new byte[bodySize];
+                                source.get(bodyBytes);
+                                try {
+                                    out.write(bodyBytes);
+                                } catch (IOException e) {
+                                    throw new UncheckedIOException(e);
+                                }
+
+                            } else {
+
+                                // Write an empty function body
+                                var bodySize = (int) readVarUInt32(source);
+                                source.position(source.position() + bodySize);
+
+                                writeVarUInt32(out, 3); // function size in bytes
+                                writeVarUInt32(out, 0); // locals count
+                                out.write(OpCode.UNREACHABLE.opcode());
+                                out.write(OpCode.END.opcode());
+                            }
```

Comment:
@andreaTP since I don't have unit tests that tests this new code path out, this section could really use a critical review.

Reply:
This has enought test coverage now.